app/uk/gov/hmrc/trusts/services/TaxEnrolmentsService.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.trusts.services
+
+import javax.inject.Inject
+
+import akka.actor.ActorSystem
+import com.google.inject.ImplementedBy
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.trusts.connector.{DesConnector, TaxEnrolmentConnector}
+import uk.gov.hmrc.trusts.exceptions.MaxRetriesAttemptedException
+import uk.gov.hmrc.trusts.models.{TaxEnrolmentFailure, TaxEnrolmentSuccess, TaxEnrolmentSuscriberResponse}
+import akka.pattern.Patterns.after
+import play.api.Logger
+import uk.gov.hmrc.trusts.config.AppConfig
+
+import scala.annotation.tailrec
+import scala.concurrent.duration._
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+class TaxEnrolmentsServiceImpl @Inject()(taxEnrolmentConnector :TaxEnrolmentConnector, config: AppConfig) extends TaxEnrolmentsService {
+
+  private val DELAY_SECONDS_BETWEEN_REQUEST = config.delayToConnectTaxEnrolment
+  private val MAX_TRIES = config.maxRetry
+
+  override def setSubscriptionId(subscriptionId: String)(implicit hc: HeaderCarrier): Future[TaxEnrolmentSuscriberResponse] = {
+    implicit val as = ActorSystem()
+    enrolSubscriberWithRetry( subscriptionId, 1)
+  }
+
+  private def enrolSubscriberWithRetry( subscriptionId: String, acc: Int)(implicit as: ActorSystem, hc: HeaderCarrier): Future[TaxEnrolmentSuscriberResponse] = {
+    makeRequest(subscriptionId) recoverWith {
+      case NonFatal(exception) => {
+        if (isMaxRetryReached(acc)) {
+          Logger.error("[enrolSubscriberWithRetry] Maximum retry completed. Tax enrolment failed.")
+          Future{TaxEnrolmentFailure}
```

Comment:
Should this be `Future.successful(TaxEnrolmentFailure)`. Currently, you are spinning up a new `Thread` within a `ThreadPool/ExecutionContext` rather than just returning a completed `Promise` with `.successful`?

Reply:
Done @adamconder